src/syntax/lexer.ml

```diff
@@ -210,27 +210,19 @@ let find_line p f =
 		loop 0 (Array.length f.lalines)
 
 (* resolve a position within a non-haxe file by counting newlines *)
-let resolve_pos file =
-	let ch = open_in_bin file in
-	let f = make_file file in
+let resolve_pos f next skip =
 	let rec loop p =
 		let inc i () =
 			f.lline <- f.lline + 1;
 			f.llines <- (p + i,f.lline) :: f.llines;
 			i
 		in
-		let i = match input_char ch with
+		let i = match next() with
 			| '\n' -> inc 1
 			| '\r' ->
-				ignore(input_char ch);
+				skip 1;
```

Comment:
Hmm... is this an assumption that this only ever shows up as `\r\n`? That seems a bit dubious even if it is true in practice...

Reply:
(And yes I know it was like this before, this PR just made me think about it.)

Reply:
I have no idea, to me it just looks like some DOS nonsense :/

I didn't want to change the behavior there because I'm worried about breaking more things than I'd fix; should we open an issue about it and fix it separately?